src/main/java/app/model/Map.java

```diff
@@ -257,4 +257,27 @@ private boolean clearSpot(Vector v)
         }
         return true;
     }
+
+
+    public void capture(Agent currentAgent)
```

Comment:
Perhaps rename to `checkForCapture` or something?  It is vague to just say "Capture" as that will make people think that it automatically captured